### PR TITLE
Added type hints, and option for bytes-like parameters

### PR DESCRIPTION
**Pull request by sambarnes**

### CODE REVIEW

File: factom/client.py [hunk 1]
```diff
@@ -151,128 +126,86 @@ def current_minute(self):
...
-        block keyMR, directory block height, and the timestamp.
+        Every directory block has a KeyMR (Key Merkle Root), which can be used to retrieve it.
+        The response will contain information that can be used to navigate through all transactions (entry
+        and factoid) within that block. The header of the directory block will contain information regarding
+        the previous directory block key merkle root, directory block height, and the timestamp.
         """
-        return self._request('directory-block', {
-            'keymr': keymr
-        })
+        return self._request("directory-block", {"keymr": keymr if type(keymr) is str else keymr.hex()})
```
Comment (sanchopansa) [inline factom/client.py]: I'm wondering if it makes sense to have this conversion function moved to a helper/util method, as it's repeated throughout pretty much the entire library now. Maybe it comes in handy if we decide to support other types of inputs in the future? (not sure how likely that is)

File: factom/client.py [hunk 2]
```diff
@@ -151,128 +126,86 @@ def current_minute(self):
...
-        Every directory block has a KeyMR (Key Merkle Root), which can
-        be used to retrieve it. The response will contain information
-        that can be used to navigate through all transactions (entry
-        and factoid) within that block. The header of the directory
-        block will contain information regarding the previous directory
-        block keyMR, directory block height, and the timestamp.
+        Every directory block has a KeyMR (Key Merkle Root), which can be used to retrieve it.
+        The response will contain information that can be used to navigate through all transactions (entry
+        and factoid) within that block. The header of the directory block will contain information regarding
+        the previous directory block key merkle root, directory block height, and the timestamp.
```
Comment (sanchopansa) [inline factom/client.py]: Minor: typo "merkle" -> "Merkle".

File: factom/client.py [hunk 3]
```diff
@@ -54,80 +52,57 @@ def url(self):
...
-        Retrieves administrative blocks for any given height.
-        The admin block contains data related to the identities within
-        the factom system and the decisions the system makes as it
-        builds the blockchain. The abentries(admin block entries) in
-        the JSON response can be of various types, the most common is a
-        directory block signature(DBSig). A majority of the federated
-        servers sign every directory block, meaning every block after
-        m5 will contain 5 DBSigs in each admin block.
+        Retrieves administrative blocks for any given height. The admin block contains data related to the identities
+        within the factom system and the decisions the system makes as it builds the blockchain.
```
Comment (sanchopansa) [inline factom/client.py]: Minor: typo "factom" -> "Factom"

File: factom/client.py [hunk 4]
```diff
@@ -151,128 +126,86 @@ def current_minute(self):
...
-        This can be used to grab the latest block and the information
+        The directory block head is the last known directory block by factom, or in other words,
+        the most recently recorded block. This can be used to grab the latest block and the information
         required to traverse the entire blockchain.
         """
-        return self._request('directory-block-head')
+        return self._request("directory-block-head")
 
-    def entry(self, hash):
+    def entry(self, hash: Union[bytes, str], encode_as_hex: bool = False):
```
Comment (sanchopansa) [inline factom/client.py]: Might be a good idea to use a different identifier for the variable instead of `hash`, as hash is also a reserved keyword.

**Review by sanchopansa — COMMENTED** [review]
Thanks for those changes, Sam! I've added some small comments, but generally LGTM.

File: factom/client.py (hunk 4 above)
Comment (sambarnes) [reply factom/client.py]: Yep good point, might as well change those names now while we're updating the functions. Done

File: factom/client.py (hunk 1 above)
Comment (sambarnes) [reply factom/client.py]: Good idea, threw a function into a utils.py file for now. Does make the client code look cleaner

**Comment by sanchopansa** [comment]
LGTM

**Pull request merged**